src/meta/api/src/schema_api_impl.rs

```diff
@@ -1071,6 +1082,222 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         Ok(index_metas)
     }
 
+    // virtual column
+
+    async fn create_virtual_column(
+        &self,
+        req: CreateVirtualColumnReq,
+    ) -> Result<CreateVirtualColumnReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+        loop {
+            trials.next().unwrap()?;
+
+            let (_, old_virtual_column_opt): (_, Option<VirtualColumnMeta>) =
+                get_pb_value(self, &req.name_ident).await?;
+
+            if old_virtual_column_opt.is_some() {
+                return Err(KVAppError::AppError(AppError::VirtualColumnAlreadyExists(
+                    VirtualColumnAlreadyExists::new(
+                        req.name_ident.table_id,
+                        format!(
+                            "create virtual column with tenant: {} table_id: {}",
+                            req.name_ident.tenant, req.name_ident.table_id
+                        ),
+                    ),
+                )));
+            }
+            let virtual_column_meta = VirtualColumnMeta {
+                table_id: req.name_ident.table_id,
+                virtual_columns: req.virtual_columns.clone(),
+                created_on: Utc::now(),
+                updated_on: None,
+            };
+
+            // Create virtual column by inserting this record:
+            // (tenant, table_id) -> virtual_column_meta
+            {
+                let if_then = vec![
+                    txn_op_put(&req.name_ident, serialize_struct(&virtual_column_meta)?), /* (tenant, table_id) -> virtual_column_meta */
+                ];
+
+                let txn_req = TxnRequest {
+                    condition: vec![],
+                    if_then,
+                    else_then: vec![],
+                };
```

Comment:
It still has to assert the seq number to be `0`. otherwise two processes will override each others' data.
I.e.:
```rust
                let condition = vec![
                    txn_cond_seq(req.name_ident, Eq, 0),
                ]
```